### PR TITLE
Add ci

### PR DESCRIPTION
**Pull request by wszdexdrf**
Branched after PR #13.

Includes tests for:
- cargo build
- cargo clippy
- cargo test using docker image containing Trezor emulator

### CODE REVIEW

**Review by danielabrozzoni — APPROVED** [review]
utACK 20f24443daae50c35d12dd34ac9e58d00065212f

Looks good to me! Can't really test if it works properly, so I'll just go ahead and merge it, and if it's broken somehow we'll fix it :) 

Also, nit for the future: commit title should start with a capital letter ("Add workflow"), and you should put in the commit description the issue that it's closing/partially fixing if there's one :)

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  pull_request:
+    branches: [ master ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
```
Comment (danielabrozzoni) [inline .github/workflows/main.yml:16]: Trailing whitespaces

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - run: cargo build
+    - uses: actions-rs/clippy-check@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        args: --all-features
+  test:
+    runs-on: ubuntu-latest
+
```
Comment (danielabrozzoni) [inline .github/workflows/main.yml:26]: Trailing whitespaces

**Review by danielabrozzoni — CHANGES_REQUESTED** [review]
While merging I actually noticed there are some trailing whitespaces, can you remove them?

**Comment by wszdexdrf** [comment]
> Also, nit for the future: commit title should start with a capital letter ("Add workflow"), and you should put in the commit description the issue that it's closing/partially fixing if there's one :)
Noted. Will remember it in the future.

**Comment by danielabrozzoni** [comment]
Since I've delayed the merge anyways, can you fix in these PR as well? :)

**Comment by wszdexdrf** [comment]
>  Can't really test if it works properly, so I'll just go ahead and merge it, and if it's broken somehow we'll fix it :)

It works on my fork, so it should work here as well. Otherwise it would be weird indeed XD.

**Comment by danielabrozzoni** [comment]
Ah ok, cool :)

**Pull request merged**